course/models/models.py

```diff
@@ -24,6 +24,7 @@ class QuestionCategory(models.Model):
     name = models.CharField(max_length=100)
     description = models.TextField()
     parent = models.ForeignKey('self', on_delete=models.SET_NULL, null=True, blank=True)
+    links_to = models.ManyToManyField('self', related_name="linked_by", symmetrical=False, blank=True)
```

Comment:
the name ('links_to', 'linked_by') gives the impression that it's a foreign key. I recommend using ('links', 'reverse_links) or ('dependents', 'dependees').
Tbh, I don't like either of them. See if you can come up with a better combination.